cf-pubsub/src/main/java/org/eclipse/californium/pubsub/Topic.java

```diff
@@ -0,0 +1,202 @@
+        this.ct = ct;
+    }
+
+    /**
+     * Creates a sorted ArrayList of object Topic that includes all of
+     * the parent Topics out of a given Set of WebLinks
+     *
+     * @param swl Set<WebLink> a Set of WebLinks
+     * @return a sorted ArrayList of Topics
+     */
+    public static ArrayList<Topic> makeArrayList(Set<WebLink> swl) {
+        ArrayList<Topic> at = new ArrayList<>();
+        for (WebLink wl : swl) {
+            Topic t = new Topic(wl);
+            t.getTopics(t, at);
+        }
+        Collections.sort(at, new TopicComparator());
+        return at;
+    }
+
+    /**
+     * Adds to a given ArrayList of Topics all of the topics and subtopics as individuals
+     *
+     * @param topic a Topic which you want to get the subtopics and parent Topics from
+     * @param at    an already existing ArrayList of Topics where you want all the Topics to be
+     */
+    public void getTopics(Topic topic, ArrayList<Topic> at) {
+
+        if (topic == null) {
+            return;
+        }
+        boolean flag = false;
+        for (Topic x :
+                at) {
+            if (topic.equals(x)) {
+                flag = true;
+            }
+        }
+        if (!flag) {
+            at.add(topic);
+        }
+        getTopics(topic.getParent(), at);
+    }
+
+    /**
+     * Prints a representation of a Topic
+     *
+     * @return String representation of Topic
+     */
+    public String toString() {
+        return this.getURIString() + "\t\t\t\t\t\t |ct: " + this.ct;
+    }
+
+    /**
+     * Returns the parent Topic of the instance Topic as an individual
+     *
+     * @return the parent topic Topic
+     */
+    public Topic getParent() {
+        String[] ps = new String[this.uri.length - 1];
+        for (int i = 0; i < this.uri.length - 1; i++) {
+            ps[i] = this.uri[i];
+        }
+        if (ps.length != 0) {
+            Topic parent = new Topic(ps, 40);
+            return parent;
+        }
+        return null;
+    }
+
+    /**
+     * @return Content Type as an int
+     */
+    public int getCt() {
+        return ct;
+    }
+
+    /**
+     * @return name of Topic as String
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * @return URI of Topic as String[]
+     */
+    public String[] getURI() {
+        return uri;
+    }
+
+    /**
+     * @return URI of Topic as String
+     */
+    public String getURIString() {
+        StringBuilder sb = new StringBuilder();
+        for (String s : this.getURI()) {
+            sb.append(s).append("/");
+        }
+        String p = sb.toString();
+        return p = p.substring(0, p.length() - 1);
+    }
+
+    /**
+     * @param topic that you want to compare
+     * @return true if given Topic has same URI as instance Topic
+     */
+    public boolean equals(Topic topic) {
+        return (this.getName().equals(topic.getName()) && this.getURIString().equals(topic.getURIString()));
```

Comment:
the name is the last part of the URI? So no need to compare it twice?

Try `Arrays.equals(this.uri, topic.uri);`